.github/workflows/build-images-base.yaml

```diff
@@ -54,7 +54,7 @@ jobs:
         uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
         with:
           persist-credentials: false
-          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
@eddie-knight Thanks for the PR. Can you explain what is the issue of using `ref: ${{ github.event.pull_request.head.sha }}`? This workflow is behind an environment, see line 23, which is only executed after being approved by a set of committers.

Reply:
Honestly I'm not able to explain much from the OpenSSF scorecard aside from what the [docs](https://github.com/ossf/scorecard/blob/7cd6406aef0b80a819402e631919293d5eb6adcf/docs/checks.md#dangerous-workflow) say 😅

We could raise an issue on the repo if it's a cocnern... but IMHO it seems like changing to just use the top commit on actions/checkout accomplishes the same thing as was here before, and we don't have to fuss with reforming the scorecard.

Reply:
@eddie-knight not specifying the right `ref`, in this case `${{ github.event.pull_request.head.sha }}`, we will default to the ref of the base branch which will not contain the changes submitted by a PR that will trigger this workflow.

Reply:
@aanm Since you're using `actions/checkout` on a `pull_request` workflow, the default ref is the "Last merge commit on the GITHUB_REF branch"

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

Reply:
@eddie-knight This workflow uses `pull_request_target`, so `GITHUB_REF` points to the PR target branch, not the PR branch, hence why we need this ;)

Reply:
Ah, I see my oversight. In that case I'll modify this PR to create a CLOMonitor exception instead.

Reply:
Ah, I did overlook that detail. So our options from here are...

1. Propose a change in the Scorecard logic
2. Create a CLOMonitor exception
3. Reconsider the use of `pull_request_target`, per the [docs recommendation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) "Avoid using this event if you need to build or run code from the pull request."

Reply:
For 3. it will mean that external contributors aren't able to use this GH workflow, which is not intended. In this case it is safe to run `pull_request_target` since this workflow is only permitted to be executed after an explicit approval of a subset of committers.

Reply:
Brilliant. I'll close out this PR and recommend you do #1 or #2 above... either one will require a bit of language you'll have to craft from your perspective.

Godspeed!

Reply:
This is fixed in #22371